src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.DocumentChangeTracker.cs

```diff
@@ -59,6 +60,20 @@ public ImmutableArray<SourceText> GetTrackedTexts()
             public LspWorkspaceManager GetLspWorkspaceManager() => _queue._lspWorkspaceManager;
 
             public bool IsComplete() => _queue._queue.IsCompleted && _queue._queue.IsEmpty;
+
+            public async Task<bool> AreAllItemsCancelledAsync()
```

Comment:
this function scares me a bit.  it asks a query question, but it mutates to process.